tests/acceptance/features/bootstrap/CapabilitiesContext.php

```diff
@@ -39,6 +40,392 @@ class CapabilitiesContext implements Context {
 	 */
 	private $featureContext;
 
+	/**
+	 * Converts a XML response into an array
+	 *
+	 * @param SimpleXMLElement $xml
+	 *
+	 * @return array
+	 */
+	public function simpleXMLElementToArray(SimpleXMLElement $xml): array {
+		$json = json_encode($xml);
+		return (array)json_decode($json, true);
+	}
+
+	/**
+	 * Converts a PyString (expected to be a valid json) to an array
+	 *
+	 * @param PyStringNode $schemaString
+	 *
+	 * @return array
+	 * @throws Exception
+	 */
+	public function getArrayFromPyString(PyStringNode $schemaString): array {
+		$schemaRawString = $schemaString->getRaw();
+		$schema = json_decode($schemaRawString, true);
+		if ($schema === null && json_last_error() !== JSON_ERROR_NONE) {
+			throw new Exception(
+				"Error: Provided JSON data is invalid." .
+				"\nMessage: '" . json_last_error_msg() . "'"
+			);
+		}
+		return (array) $schema;
+	}
+	/**
+	 * Determines if the given array is multidimensional or not
+	 *
+	 * @param array $arrayToCheck
+	 *
+	 * @return bool
+	 */
+	public function isMultiDimensionalArray(array $arrayToCheck): bool {
+		if (\count($arrayToCheck) === 0) {
+			return false;
+		}
+		// The XML converted array can have only values but may be faked as multidimensional
+		// by having a single key named 'element' with the value being an array or a string
+		// This function checks if the array has only one key named 'element'
+		// if 'element' is as the array key, then it is not a multidimensional array
+		$keys = \array_keys($arrayToCheck);
+		if ($keys === ["element"]) {
+			return false;
+		}
+
+		$values = \array_values($arrayToCheck);
+
+		// rsort will sort the array in descending order
+		// if any values in the array are arrays, then the first element will be an array
+		// if the first element is an array, then it is a multidimensional array
+		\rsort($values);
+		return \is_array($values[0]);
+	}
+
+	/**
+	 * Array converted from XML can have a different structure.
+	 *
+	 * For eg. a 1D array can be like: Array(
+	 * 		[element] => Array(
+	 * 			[0] => shareExpiration,
+	 * 			[1] => passwordProtected
+	 * 		)
+	 * )
+	 *
+	 * or like: Array(
+	 * 		[0] => shareExpiration
+	 * )
+	 *
+	 * If element is the only key in the array, then we need to get the value of that key
+	 * and return it.
+	 *
+	 * @param array $arr the array to grab the value from
+	 *
+	 * @return array array of values of the provided array
+	 */
+	public function getArrayValues(array $arr): array {
+		$values = [];
+		if (\array_keys($arr) === ["element"]) {
+			if (\is_array($arr["element"])) {
+				$values = $arr["element"];
+			} else {
+				$values = [$arr["element"]];
+			}
```

Comment:
```suggestion
			$values = (\is_array($arr["element"])) ? $arr["element"] : [$arr["element"]];
```
Prefer to use ternary op.